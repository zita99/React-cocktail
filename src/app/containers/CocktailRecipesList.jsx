import React from 'react';
import CocktailRecipe from '../components/CocktailRecipe.jsx';
import { connect } from "react-redux";
import { showImage } from "../actions/fullImageActions";

export class CocktailRecipesList extends React.Component {

  render() {
    if (this.props.loading) {
      return null;
    }
    if (!this.props.cocktailList) {
        return (
          <section id='recipes-list'>
            <div className='container recipes-list_not-found'>
              <p>Ooops it looks like we don't have this cocktail recipe for you...</p>
              <p>Why not try another search?</p>
            </div>
          </section>
        );
      } else {
        const recipes = this.props.cocktailList.map((recipe) => {
        return <CocktailRecipe key={ recipe.idDrink } id={ recipe.idDrink } recipe={ recipe } showFullImage={ this.props.showImage }/>
      });
      return (
        <section id='recipes-list'>
          <div className='container recipes-list'>
            <h1 className='col-12 recipes-list_heading'>Random drinks 0.1</h1>
            { recipes }
          </div>
        </section>
      );
    }
  }
}

const mapStateToProps = (state) => {
  return {
    cocktailList: state.recipesListReducer.cocktailList,
    loading: state.recipesListReducer.loading
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
      showImage: (url, details) => {
        dispatch(showImage(url, details));
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CocktailRecipesList);
