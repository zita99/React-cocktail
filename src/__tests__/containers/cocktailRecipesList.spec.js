import React from 'react';
import { CocktailRecipesList } from '../../app/containers/CocktailRecipesList.jsx';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('CocktailRecipesList', () => {

  it('should not render if loading is true', () => {

    const wrapper = shallow(<CocktailRecipesList loading = { true }/>);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should render recipes list not found if fetched cocktail list is undefinded', () => {

    const testCocktailList = undefined;

    const wrapper = shallow(<CocktailRecipesList cocktailList = { undefined }/>);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

  it('should render recipes list if fetched cocktail list is not empty', () => {

    const testCocktailList = [{idDrink: 'cocktail - 1'}, {idDrink: 'cocktail - 2'}, {idDrink: 'cocktail - 3'}];

    const wrapper = shallow(<CocktailRecipesList cocktailList = { testCocktailList }/>);

    expect(toJson(wrapper)).toMatchSnapshot();

  });

});
