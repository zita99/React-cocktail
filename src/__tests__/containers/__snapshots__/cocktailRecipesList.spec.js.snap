// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailRecipesList should not render if loading is true 1`] = `null`;

exports[`CocktailRecipesList should render recipes list if fetched cocktail list is not empty 1`] = `
<section
  id="recipes-list"
>
  <div
    className="container recipes-list"
  >
    <h1
      className="col-12 recipes-list_heading"
    >
      Random drinks 0.1
    </h1>
    <CocktailRecipe
      id="cocktail - 1"
      key="cocktail - 1"
      recipe={
        Object {
          "idDrink": "cocktail - 1",
        }
      }
    />
    <CocktailRecipe
      id="cocktail - 2"
      key="cocktail - 2"
      recipe={
        Object {
          "idDrink": "cocktail - 2",
        }
      }
    />
    <CocktailRecipe
      id="cocktail - 3"
      key="cocktail - 3"
      recipe={
        Object {
          "idDrink": "cocktail - 3",
        }
      }
    />
  </div>
</section>
`;

exports[`CocktailRecipesList should render recipes list not found if fetched cocktail list is undefinded 1`] = `
<section
  id="recipes-list"
>
  <div
    className="container recipes-list_not-found"
  >
    <p>
      Ooops it looks like we don't have this cocktail recipe for you...
    </p>
    <p>
      Why not try another search?
    </p>
  </div>
</section>
`;
